### PR TITLE
feat(cta-interest): implemented proper events

### PR DESCRIPTION
**Pull request by milan-bc**
## Description (optional)
This PR brings on proper matomo events

## Testing Steps (optional)
Use gold tier 2 account, go over SB, once when there is a interest banner you can confirm in net tab that event has been dispatched

### CODE REVIEW

File: packages/blockchain-wallet-v4-frontend/src/modals/SimpleBuy/OrderSummary/InterestBanner/template.success.tsx [hunk 1]
```diff
@@ -55,9 +57,10 @@ const InterestBanner: React.FC<Props> = ({
         style={{ marginTop: '16px', maxWidth: '144px' }}
         nature='light'
         data-e2e='earnInterestNow'
-        onClick={() =>
+        onClick={() => {
```
Comment (pedroapfilho) [inline packages/blockchain-wallet-v4-frontend/src/modals/SimpleBuy/OrderSummary/InterestBanner/template.success.tsx:60]: Wouldn't be better to create an external function to do it?

File: packages/blockchain-wallet-v4-frontend/src/data/components/interest/sagas.ts [hunk 2]
```diff
@@ -377,6 +377,20 @@ export default ({
...
         actions.analytics.logEvent(INTEREST_EVENTS.DEPOSIT.SEND_SUCCESS)
       )
+
+      const afterTransactionR = yield select(
+        selectors.components.interest.getAfterTransaction
+      )
+      const afterTransaction = afterTransactionR.getOrElse({
+        show: false
+      } as InterestAfterTransactionType)
+      if (afterTransaction && afterTransaction.show) {
```
Comment (pedroapfilho) [inline packages/blockchain-wallet-v4-frontend/src/data/components/interest/sagas.ts:387]: idk if you want to check both, but you could do: 
```suggestion
      if (afterTransaction?.show) {
```

Comment (milan-bc) [reply packages/blockchain-wallet-v4-frontend/src/data/components/interest/sagas.ts:387]: yeah forgot to use this way, will update ;)

File: packages/blockchain-wallet-v4-frontend/src/modals/SimpleBuy/OrderSummary/InterestBanner/template.success.tsx (hunk 1 above)
Comment (milan-bc) [reply packages/blockchain-wallet-v4-frontend/src/modals/SimpleBuy/OrderSummary/InterestBanner/template.success.tsx:60]: we can bring this to tech debt, I do share your opinion, from some reason we were more into this regime to put things directly without external functions, that is why I do follow it in a first place.

Comment (hanabi62513) [reply packages/blockchain-wallet-v4-frontend/src/modals/SimpleBuy/OrderSummary/InterestBanner/template.success.tsx:60]: 誰やねんお前

File: packages/blockchain-wallet-v4-frontend/src/data/components/interest/sagas.ts (hunk 2 above)
Comment (hanabi62513) [reply packages/blockchain-wallet-v4-frontend/src/data/components/interest/sagas.ts:387]: 何をしたいの私からお金を巻き上げ落としてるわけたりみんな 英語分からないんだけど私

**Comment by hanabi62513** [comment]
このサイトは何で私のメールアドちっちゃいは何がしたいかわからんのほんま

2021年2月19日(金) 0:33 Milan <notifications@github.com>:

**Review by schnogz — APPROVED** [review]

**Pull request merged**